### DISCUSSION TITLE
Feature Request: Add role permissions to hide download menu item from dashboard view

### DISCUSSION THREAD

**Discussion by raunaksinha7ai** [apache/superset#30685] [Ideas]
We are looking for guidance on how to hide the "Download as Image" and "Export to PDF" options using role-based permissions in Superset.

We need to ensure that only users with specific assigned roles are able to download or export data from dashboards. While we have successfully restricted other download/export options through permissions, we are struggling to apply the same restrictions for these two features.

Is there a way to achieve this within the current permission framework, or would it require a custom solution?

**Comment by naganeuron7** [comment]
We are also facing similar challenge; appreciate if anyone have any ideas.

Thanks in advance.
